sympy/parsing/latex/lark/latex_parser.py

```diff
@@ -0,0 +1,310 @@
+            }
+            op, left = args[0].type, args[1]
+            print('unary', op, left)
+            return op_map[op](left)
+
+        def factorial(self, args):
+            # ensure we handle x!!!!
+            result = sympy.factorial(args[0])
+            for _ in range(len(args) - 2):
+                result = sympy.factorial(result)
+            return result
+
+        def limit(self, args):
+            sub = args[1].children
+            if len(sub) == 9: # e.g. \\lim_{x \\to 3^{+}} a
+                direction = '+' if sub[7].type == 'ADD' else '-'
+                return sympy.Limit(args[2], sub[2], sub[4], dir=direction)
+            else: # e.g. \\lim_{x \\to 3} a
+                return sympy.Limit(args[2], sub[2], sub[4])
+
+        def binary_op_1(self, args):
+            print('op_1', args)
+            return self._binary_op(args)
+
+        def binary_op_2(self, args):
+            print('op_2', args)
+            return self._binary_op(args)
+
+        def binary_op_3(self, args):
+            left, op, right = args[0], args[1][0], args[1][1]
+            print('op_3', op, left, right)
+            return self._binary_op([left, op, right])
+
+        def sup_expression(self, args):
+            if len(args) == 2: # ^ atom form
+                return args
+            else: # ^ { atom } form
+                return [args[0], args[2]]
+
+        def _binary_op(self, args):
+            op_map = {
+                'ADD': operator.add,
+                'SUB': operator.sub,
+                'MUL': operator.mul,
+                'CMD_TIMES': operator.mul,
+                'CMD_CDOT': operator.mul,
+                'DIV': operator.truediv,
+                'CMD_DIV': operator.truediv,
+                'CARET': pow
+            }
+            left, op, right = args[0], args[1].type, args[2]
+            result = op_map[op](left, right)
+            for i in range((len(args) - 3) // 2):
+                left, op, right = result, args[i*2 + 3].type, args[i*2 + 4]
+                result = op_map[op](left, right)
+            return result
+
+        def fraction(self, args):
+            return args[2] * sympy.Pow(args[5], -1)
+
+        def summation(self, args):
+            if hasattr(args[1], 'children'):
+                sub, sup = args[1].children, args[2]
+            else:
+                sup, sub = args[1], args[2].children
+            return sympy.Sum(args[3], (sub[2], sub[4], sup[1]))
+
+        def product(self, args):
+            if hasattr(args[1], 'children'):
+                sub, sup = args[1], args[2]
+            else:
+                sup, sub = args[1], args[2]
+            return sympy.Product(args[3], (sub.children[2], sub.children[4], sup[1]))
+
+        def sqrt(self, args):
+            if len(args) == 7: # e.g \\sqrt [ 2 ] { a }
+                return sympy.root(args[5], args[2])
+            else: # e.g. \\sqrt { a }
+                return sympy.sqrt(args[2])
```

Comment:
It could be better to split the rules in the grammar than defining the conditions here.
I often try to minimize the conditions in transformer, if it can be done with the grammar itself, for better practice, and hopefully the library can optimize that part.